pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
```

Comment:
@mikedanese HeldBy is a required field so I don't like the idea of putting into a field that can be optional (AcquireInfo).  I'm fine adding the AcquireInfo field if that is desired, but HeldBy shouldn't be put in there.